packages/format-library/src/collab-comment/collab-board/index.js

```diff
@@ -0,0 +1,538 @@
+	const popoverAnchor = useAnchor( {
+		editableContentElement: contentRef.current,
+	} );
+
+	// State to manage the comment thread.
+	const [ inputComment, setInputComment ] = useState( '' );
+	const [ isResolved, setIsResolved ] = useState( false );
+	const [ isEditing, setIsEditing ] = useState( null );
+	const [ showConfirmation, setShowConfirmation ] = useState( false );
+	const [ threadId, setThreadId ] = useState( null );
+
+	// Get the dispatch functions to save the comment and update the block attributes.
+	const { saveEntityRecord } = useDispatch( coreStore );
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	// Set the threadId if exists, from the currently selected block classList.
+	useEffect( () => {
+		const classList = contentRef.current?.classList?.value
+			.split( ' ' )
+			.find( ( className ) =>
+				className.startsWith( 'block-editor-collab__' )
+			);
+
+		setThreadId(
+			classList
+				? classList.slice( 'block-editor-collab__'.length )
+				: Date.now()
+		);
+	}, [ contentRef ] );
+
+	// Add border to the block if threadId exists.
+	useEffect( () => {
+		if ( threadId ) {
+			addBorder();
+		}
+	}, [ threadId ] );
+
+	// Fetch the current post, current user, and the selected block clientId.
+	const {
+		postId,
+		allThreads,
+		currentThread,
+		currentUser,
+		clientId,
+		commentsCount,
+		isCurrentThreadResolved,
+	} = useSelect(
+		( select ) => {
+			const post = select( editorStore ).getCurrentPost();
+			const collabData = post?.meta?.collab
+				? JSON.parse( post.meta.collab )
+				: [];
+
+			const thread = collabData[ threadId ] ?? {};
+			const threadIsResolved = thread.threadIsResolved || false;
+			const count = threadIsResolved ? 0 : thread.comments?.length || 0;
+
+			return {
+				postId: post?.id,
+				allThreads: collabData,
+				currentThread: thread,
+				commentsCount: count,
+				isCurrentThreadResolved: threadIsResolved,
+				currentUser: select( coreStore ).getCurrentUser()?.name || null,
+				clientId:
+					select( blockEditorStore ).getSelectedBlockClientId() ||
+					null,
+			};
+		},
+		[ threadId ]
+	);
+
+	// Helper function to generate a new comment.
+	const generateNewComment = () => ( {
+		commentId: Date.now(),
+		createdBy: currentUser,
+		comment: inputComment,
+		createdAt: new Date().toISOString(),
+	} );
+
+	// Helper function to get updated comments structure
+	const getUpdatedComments = ( newComment, threadKey ) => ( {
+		...allThreads,
+		[ threadKey ]: {
+			isResolved,
+			createdAt:
+				allThreads?.threadKey?.createdAt || new Date().toISOString(),
+			createdBy: currentUser,
+			comments: [
+				...( allThreads[ threadKey ]?.comments || [] ),
+				newComment,
+			],
+		},
+	} );
+
+	// Function to save the comment.
+	const saveComment = async () => {
+		const newComment = generateNewComment();
+		const updatedComments = getUpdatedComments( newComment, threadId );
+
+		await saveEntityRecord( 'postType', 'post', {
```

Comment:
Should we use `useEntityProp` instead? I'm guessing that this is the source of [problem](https://github.com/WordPress/gutenberg/pull/60622#issuecomment-2173311846) I'm having.

Reply:
Done!